tests/organizations/test_get_program.py

```diff
@@ -0,0 +1,304 @@
+        
+        mock_get_response = Mock()
+        mock_get_response.json.return_value = expected_user
+        mock_get_response.status_code = success_code
+
+        mock_get_user.return_value = mock_get_response
+        mock_get_user.raise_for_status = json.dumps(success_code)
+        
+        user_login_success = {
+            "username": user1.get("username"),
+            "password": user1.get("password")
+        }
+        
+        with self.client:
+            login_response = self.client.post(
+                "/login",
+                data=json.dumps(user_login_success),
+                follow_redirects=True,
+                content_type="application/json",
+            )
+
+        test_user1 = UserModel(
+            name=user1["name"],
+            username=user1["username"],
+            password=user1["password"], 
+            email=user1["email"], 
+            terms_and_conditions_checked=user1["terms_and_conditions_checked"]
+        )
+        test_user1.need_mentoring = user1["need_mentoring"]
+        test_user1.available_to_mentor = user1["available_to_mentor"]
+
+        test_user1.save_to_db()
+        self.test_user1_data = UserModel.find_by_email(test_user1.email)
+        AUTH_COOKIE["user"] = marshal(self.test_user1_data, full_user_api_model)
+
+        test_user_extension = UserExtensionModel(
+            user_id=self.test_user1_data.id,
+            timezone="AUSTRALIA_MELBOURNE"
+        )
+        test_user_extension.is_organization_rep = True
+        test_user_extension.save_to_db()
+
+        # prepare existing organization
+        # created on the 2020-09-30 10:00 AEST+1000
+        organization = OrganizationModel(
+            rep_id=self.test_user1_data.id, 
+            name="Company ABC",
+            email="companyabc@mail.com",
+            address="506 Elizabeth St, Melbourne VIC 3000, Australia",
+            website="https://www.ames.net.au",
+            timezone="AUSTRALIA_MELBOURNE",
+        )
+        organization.rep_department = "H&R Department"
+        organization.about = "This is about ABC"
+        organization.phone = "321-456-789"
+        organization.status = "DRAFT"
+        organization.join_date = 1601424000
+
+        organization.save_to_db()
+        self.organization1_data = OrganizationModel.find_by_representative(self.test_user1_data.id)
+        
+        # prepare expected representative object
+        self.expected_representative = marshal(self.test_user1_data, public_user_personal_details_response_model)
+
+        
+    @patch("requests.get")
+    def test_api_dao_get_program_successfully(self, mock_get_representative):
+        # prepare existing program with 
+        # start date 2020-10-15 12:00 GMT+0
+        # and end date 2020-11-25 12:00 GMT+0
+        program = ProgramModel(
+            program_name="Program A",
+            start_date=1602763200,
+            end_date=1606305600,
+            organization_id=self.organization1_data,
```

Comment:
Just want to ask why organization1_data is used instead of organization1_data.id

Reply:
@KapilBansal320, this is because Organization to Program has one-to-many relationship where as other tables relation (user to organization, user to user_extension, user to personal_background) has one-to-one relationship. If you try put `id` behind `self.organization1_data`, you will get this error
<img width="775" alt="Screen Shot 2020-09-16 at 11 12 10 am" src="https://user-images.githubusercontent.com/29667122/93280649-7df9af00-f80d-11ea-95c9-56faf90cf277.png">

and when you look up online, this was mentioned in the [stackoverflow discussion](https://stackoverflow.com/questions/16151729/attributeerror-int-object-has-no-attribute-sa-instance-state) as being the cause of error.